pool/pool.go

```diff
@@ -285,31 +294,49 @@ func (x *wrapperPrm) setResponseInfoCallback(f func(sdkClient.ResponseMetaInfo)
 	x.responseInfoCallback = f
 }
 
+// setStatisticCallback set callback for external statistic.
+func (x *wrapperPrm) setStatisticCallback(statisticCallback stat.OperationCallback) {
+	x.statisticCallback = statisticCallback
+}
+
 // getNewClient returns a new [sdkClient.Client] instance using internal parameters.
-func (x *wrapperPrm) getNewClient() (*sdkClient.Client, error) {
+func (x *wrapperPrm) getNewClient(statisticCallback stat.OperationCallback) (*sdkClient.Client, error) {
 	var prmInit sdkClient.PrmInit
 	prmInit.SetDefaultSigner(x.signer)
 	prmInit.SetResponseInfoCallback(x.responseInfoCallback)
+	prmInit.SetStatisticCallback(statisticCallback)
 
 	return sdkClient.New(prmInit)
 }
 
 // newWrapper creates a clientWrapper that implements the client interface.
 func newWrapper(prm wrapperPrm) (*clientWrapper, error) {
-	cl, err := prm.getNewClient()
-	if err != nil {
-		return nil, err
-	}
-
 	res := &clientWrapper{
-		client:              cl,
 		clientStatusMonitor: newClientStatusMonitor(prm.address, prm.errorThreshold),
 		prm:                 prm,
+		statisticCallback:   prm.statisticCallback,
 	}
 
+	// integrate clientWrapper middleware to handle errors and wrapped client health.
+	cl, err := prm.getNewClient(res.statisticMiddleware)
+	if err != nil {
+		return nil, err
+	}
+
+	res.client = cl
+
 	return res, nil
 }
 
+func (c *clientWrapper) statisticMiddleware(nodeKey []byte, endpoint string, method stat.Method, duration time.Duration, err error) {
+	c.incRequests(duration, MethodIndex(method))
```

Comment:
This one won't be needed after default statistics decoupling from the pool.

Reply:
Still there and we still have `Statistics` which can be reworked into a `stat` thing (some structure that exposes the same interface to retrieve statistics and implements `OperationCallback`).

Reply:
Moved to `stat` package